### PR TITLE
Replacing place_items with place_loot

### PR DESCRIPTION
**Pull request by Mshock777**
Nothing special, you might want to check numbers just in case.

### CODE REVIEW

File: data/json/mapgen/house08.json [hunk 1]
```diff
@@ -85,21 +85,21 @@
...
-                { "item" : "kitchen",    "chance" : 50, "x" : [21,21], "y" : [11,11] },
-                { "item" : "kitchen",    "chance" : 50, "x" : [18,18], "y" : [11,11] },
-                { "item" : "kitchen",    "chance" : 70, "x" : [18,20], "y" : [ 7,10] },
-                { "item" : "oven",       "chance" : 50, "x" : [21,21], "y" : [10,10] },
-                { "item" : "cleaning",   "chance" : 50, "x" : [ 2, 5], "y" : [13,14] },
-                { "item" : "softdrugs",  "chance" : 70, "x" : [ 2, 5], "y" : [12,14] },
-                { "item" : "office",     "chance" : 50, "x" : [ 2, 7], "y" : [ 2, 4] }
+            "place_loot" : [
+                { "group" : "bed",        "chance" : 90, "x" : [ 4, 5], "y" : [ 7, 8], "repeat" : [1,4] },
+                { "group" : "bedroom",    "chance" : 80, "x" : [ 2, 8], "y" : [ 6, 8], "repeat" : [1,6] },
```
Comment (mugling) [inline data/json/mapgen/house08.json]: These numbers are too high. It would be worth scaling them down somewhat

Comment (Mshock777) [inline data/json/mapgen/house08.json]: Can you tell me what groups are you worried about the most?

Comment (mugling) [inline data/json/mapgen/house08.json]: I'd go down with most numbers by about 30%. The best way to think about it is that a house should contain (on average) no more than a dozen items or so

**Comment by Mshock777** [comment]
What do you think about decreasing the rate at which pot spawns? Apart from volume, it's significant upgrade, but now they both spawn at the same rate.

File: data/json/mapgen/house05.json [hunk 2]
```diff
@@ -72,31 +72,31 @@
...
+            "place_loot" : [
+                { "group" : "bed",                  "chance" : 90, "x" : [ 2, 3], "y" : [18,19], "repeat" : [1,3] },
+                { "group" : "bed",                  "chance" : 90, "x" : [12,13], "y" : [18,19], "repeat" : [1,3] },
+                { "group" : "livingroom",           "chance" : 90, "x" : [ 1, 8], "y" : [ 2,10], "repeat" : [1,4] },
+                { "group" : "livingroom",           "chance" : 90, "x" : [ 1,21], "y" : [12,15], "repeat" : [1,4] },
+                { "group" : "bedroom",              "chance" : 80, "x" : [ 1,10], "y" : [17,21], "repeat" : [1,4] },
+                { "group" : "dresser",              "chance" : 85, "x" : [ 2, 2], "y" : [17,17], "repeat" : [1,3] },
+                { "group" : "bedroom",              "chance" : 80, "x" : [12,21], "y" : [17,21], "repeat" : [1,4] },
+                { "group" : "consumer_electronics", "chance" : 50, "x" : [12,21], "y" : [17,21], "repeat" : [1,3] },
+                { "group" : "homeguns",             "chance" :  5, "x" : [ 1,10], "y" : [17,21] },
```
Comment (mugling) [inline data/json/mapgen/house05.json]: I'd avoid `homeguns`, its a throwback from the old system and remains only for compatibility. You want `guns_common` instead

File: data/json/mapgen/house05.json [hunk 3]
```diff
@@ -72,31 +72,31 @@
...
+                { "group" : "dresser",              "chance" : 85, "x" : [21,21], "y" : [21,21], "repeat" : [1,3] },
+                { "group" : "novels",               "chance" : 80, "x" : [ 8, 8], "y" : [ 3, 5], "repeat" : [1,3] },
+                { "group" : "novels",               "chance" : 80, "x" : [13,13], "y" : [10,12], "repeat" : [1,4] },
+                { "group" : "cleaning",             "chance" : 90, "x" : [10,13], "y" : [ 3, 5], "repeat" : [1,2] },
+                { "group" : "softdrugs",            "chance" : 75, "x" : [10,13], "y" : [ 3, 5], "repeat" : [1,4] },
```
Comment (mugling) [inline data/json/mapgen/house05.json]: Too high. This group contains `disinfectant` and other such good loot

File: data/json/mapgen/house06.json [hunk 4]
```diff
@@ -87,31 +87,31 @@
...
+                { "group" : "bedroom",              "chance" : 80, "x" : [ 1, 6], "y" : [15,21], "repeat" : [1,4] },
+                { "group" : "consumer_electronics", "chance" : 50, "x" : [ 1, 6], "y" : [15,21], "repeat" : [1,3] },
+                { "group" : "homeguns",             "chance" :  5, "x" : [ 2, 2], "y" : [21,21] },
+                { "group" : "dresser",              "chance" : 80, "x" : [ 2, 2], "y" : [20,20], "repeat" : [1,2] },
+                { "group" : "dresser",              "chance" : 80, "x" : [ 2, 2], "y" : [17,17], "repeat" : [1,4] },
+                { "group" : "bedroom",              "chance" : 80, "x" : [15,21], "y" : [17,21], "repeat" : [1,4] },
+                { "group" : "dresser",              "chance" : 80, "x" : [20,20], "y" : [21,21], "repeat" : [1,2] },
+                { "group" : "dresser",              "chance" : 80, "x" : [17,17], "y" : [21,21], "repeat" : [1,2] },
+                { "group" : "cleaning",             "chance" : 90, "x" : [ 7,14], "y" : [17,21], "repeat" : [1,2] },
+                { "group" : "softdrugs",            "chance" : 75, "x" : [ 7,14], "y" : [17,21], "repeat" : [1,4] },
```
Comment (mugling) [inline data/json/mapgen/house06.json]: Again too high, same reason

File: data/json/mapgen/house06.json [hunk 5]
```diff
@@ -87,31 +87,31 @@
...
-                { "item" : "stash_food",           "chance" : 40, "x" : [16,16], "y" : [ 2, 5] },
-                { "item" : "pasta",                "chance" : 40, "x" : [20,20], "y" : [ 2, 5] },
-                { "item" : "stash_food",           "chance" : 40, "x" : [20,20], "y" : [ 2, 5] }
+            "place_loot" : [
+                { "group" : "bed",                  "chance" : 90, "x" : [ 2, 3], "y" : [18,19], "repeat" : [1,3] },
+                { "group" : "bed",                  "chance" : 90, "x" : [18,19], "y" : [20,21], "repeat" : [1,3] },
+                { "group" : "livingroom",           "chance" : 90, "x" : [ 1,10], "y" : [ 8,13], "repeat" : [1,5] },
+                { "group" : "bedroom",              "chance" : 80, "x" : [ 1, 6], "y" : [15,21], "repeat" : [1,4] },
+                { "group" : "consumer_electronics", "chance" : 50, "x" : [ 1, 6], "y" : [15,21], "repeat" : [1,3] },
+                { "group" : "homeguns",             "chance" :  5, "x" : [ 2, 2], "y" : [21,21] },
```
Comment (mugling) [inline data/json/mapgen/house06.json]: `guns_common`. There's plenty of ammo from monsterdrops, gun basements, gun shops etc

File: data/json/mapgen/house09.json [hunk 6]
```diff
@@ -83,36 +83,34 @@
...
+                { "group" : "livingroom",           "chance" : 90, "x" : [12,21], "y" : [ 2, 6], "repeat" : [1,4] },
+                { "group" : "livingroom",           "chance" : 90, "x" : [ 2,10], "y" : [ 6,11], "repeat" : [1,4] },
+                { "group" : "dining",               "chance" : 85, "x" : [16,18], "y" : [ 8, 9], "repeat" : [1,3] },
+                { "group" : "cannedfood",           "chance" : 80, "x" : [20,21], "y" : [11,11], "repeat" : [1,2] },
+                { "group" : "dining",               "chance" : 90, "x" : [20,21], "y" : [11,11], "repeat" : [1,2] },
+                { "group" : "bedroom",              "chance" : 80, "x" : [ 1,10], "y" : [17,21], "repeat" : [1,5] },
+                { "group" : "dresser",              "chance" : 80, "x" : [ 2, 2], "y" : [17,17], "repeat" : [1,3] },
+                { "group" : "dresser",              "chance" : 80, "x" : [ 2, 2], "y" : [20,20], "repeat" : [1,3] },
+                { "group" : "consumer_electronics", "chance" : 60, "x" : [ 1,10], "y" : [17,21], "repeat" : [1,3] },
+                { "group" : "homeguns",             "chance" : 10, "x" : [ 1,10], "y" : [17,21] },
```
Comment (mugling) [inline data/json/mapgen/house09.json]: Again `guns_common` or `guns_pistol_common` (typically the latter should be preferred)

File: data/json/mapgen/house09.json [hunk 7]
```diff
@@ -83,36 +83,34 @@
...
+                { "group" : "novels",               "chance" : 85, "x" : [ 2, 2], "y" : [ 8,10], "repeat" : [1,3] },
+                { "group" : "softdrugs",            "chance" : 75, "x" : [15,15], "y" : [15,15], "repeat" : [1,4] },
```
Comment (mugling) [inline data/json/mapgen/house09.json]: A little less, same reason

File: data/json/mapgen/house10.json [hunk 8]
```diff
@@ -81,29 +81,29 @@
...
-                { "item" : "bedroom",    "chance" : 60, "x" : [14,20], "y" : [17,20] }
+            "place_loot" : [
+                { "group" : "dresser",    "chance" : 80, "x" : [18,18], "y" : [20,20], "repeat" : [1,4] },
+                { "group" : "dresser",    "chance" : 80, "x" : [20,20], "y" : [ 8, 8], "repeat" : [1,4] },
+                { "group" : "dresser",    "chance" : 90, "x" : [16,16], "y" : [13,13], "repeat" : [1,2] },
+                { "group" : "dresser",    "chance" : 90, "x" : [16,16], "y" : [15,15], "repeat" : [1,2] },
+                { "group" : "dining",     "chance" : 85, "x" : [ 6, 7], "y" : [ 5, 7], "repeat" : [1,3] },
+                { "group" : "jackets",    "chance" : 70, "x" : [ 3, 3], "y" : [13,13], "repeat" : [1,2] },
+                { "group" : "homebooks",  "chance" : 70, "x" : [ 6, 6], "y" : [14,14], "repeat" : [1,3] },
+                { "group" : "homeguns",   "chance" :  4, "x" : [16,20], "y" : [ 8,11] },
```
Comment (mugling) [inline data/json/mapgen/house10.json]: Same again

File: data/json/mapgen/house11.json [hunk 9]
```diff
@@ -85,27 +85,27 @@
...
+                { "group" : "jackets",              "chance" : 60, "x" : [ 3, 3], "y" : [10,10], "repeat" : [1,2] },
+                { "group" : "softdrugs",            "chance" : 75, "x" : [ 3, 4], "y" : [12,15], "repeat" : [1,4] },
+                { "group" : "cleaning",             "chance" : 90, "x" : [ 3, 4], "y" : [12,15], "repeat" : [1,2] },
+                { "group" : "harddrugs",            "chance" : 15, "x" : [ 3, 4], "y" : [12,15], "repeat" : [1,2] },
+                { "group" : "fridge",               "chance" : 90, "x" : [ 7, 7], "y" : [20,20], "repeat" : [1,5] },
+                { "group" : "kitchen",              "chance" : 80, "x" : [ 3, 7], "y" : [17,20], "repeat" : [1,4] },
+                { "group" : "oven",                 "chance" : 75, "x" : [ 5, 5], "y" : [20,20], "repeat" : [1,3] },
+                { "group" : "livingroom",           "chance" : 90, "x" : [ 6,12], "y" : [10,15], "repeat" : [1,4] },
+                { "group" : "bedroom",              "chance" : 80, "x" : [ 3, 7], "y" : [ 4, 8], "repeat" : [1,5] },
+                { "group" : "homeguns",             "chance" :  5, "x" : [ 3, 7], "y" : [ 4, 8] },
```
Comment (mugling) [inline data/json/mapgen/house11.json]: `guns_pistol_common`

**Comment by mugling** [comment]
> Nothing special, you might want to check numbers just in case.

Good apart from `softdrugs` and `homeguns`

**Comment by Mshock777** [comment]
Please look at my second comment.

**Comment by mugling** [comment]
> What do you think about decreasing the rate at which pot spawns? Apart from volume, it's significant upgrade, but right now they both spawn at the same rate.

I'm not sure on that front. `volume` is quite a significant penalty in the early game. It might be worth making that suggestion in #16333.

**Comment by Mshock777** [comment]
Check out prepper house - that's an important one.

File: data/json/mapgen/house_garage_prepper.json [hunk 10]
```diff
@@ -101,26 +101,26 @@
...
-                { "item" : "guns_survival",  "chance" :  70, "x" : [15,16], "y" : [ 8, 9], "magazine" : 100, "ammo" : 75 },
-                { "item" : "fridge",         "chance" :  50, "x" : [ 4, 4], "y" : [11,11] },
-                { "item" : "oven",           "chance" :  50, "x" : [ 3, 3], "y" : [11,11] },
-                { "item" : "softdrugs",      "chance" :  75, "x" : [ 2, 4], "y" : [13,16] },
-                { "item" : "cleaning",       "chance" :  75, "x" : [ 2, 4], "y" : [13,16] }
+            "place_loot": [
+                { "group" : "bed",            "chance" : 95, "x" : [13,14], "y" : [20,21], "repeat" : [1,3] },
+                { "group" : "bedroom",        "chance" : 85, "x" : [ 6,21], "y" : [13,20], "repeat" : [1,4] },
+                { "group" : "livingroom",     "chance" : 90, "x" : [ 1,12], "y" : [ 2, 6], "repeat" : [1,5] },
+                { "group" : "guns_common",    "chance" : 75, "x" : [12,12], "y" : [21,21], "magazine" : 100, "ammo" : 100 },
```
Comment (mugling) [inline data/json/mapgen/house_garage_prepper.json]: Possibly `guns_survival` here as well?

Comment (Mshock777) [inline data/json/mapgen/house_garage_prepper.json]: Hmm, I'm not sure, but if you want to...
Is it everything for this pr?

Comment (mugling) [inline data/json/mapgen/house_garage_prepper.json]: It's not essential by any means. I can't see any other problems so I'll try merge-testing it.

**Comment by mugling** [comment]
Good work btw

**Comment by Mshock777** [comment]
Nah, you joined few months and already have 3 time more pr than me.
I wish I had more spare time to continue learning C to help out with code.

**Comment by Mshock777** [comment]
Btw, are you planning on reworking item spawn scaling with place_loot before 0.D?

**Comment by mugling** [comment]
Yes

**Pull request merged**